Dockerfile

```diff
@@ -0,0 +1,20 @@
+# stage 1 Generate celestia-appd Binary
+FROM golang:1.17-alpine as builder
+RUN apk update && apk --no-cache add make gcc git musl-dev
+COPY . /celestia-app
+WORKDIR /celestia-app
+RUN make build
+
+# stage 2
+FROM alpine
+RUN apk update && apk --no-cache add bash
+
+COPY --from=builder /celestia-app/build/celestia-appd /bin/celestia-appd
+COPY  docker/entrypoint.sh /opt/entrypoint.sh
+
+# p2p, rpc and prometheus port
+EXPOSE 26656 26657 1317 9090
+
+ENV CELESTIA_HOME /opt
+
+ENTRYPOINT [ "/bin/bash", "/opt/entrypoint.sh" ]
```

Comment:
I'd prefer
```
ENTRYPOINT ["/opt/entrypoint.sh"]
CMD ["/bin/celestia-appd"]
```
and the suggested changes in entrypoint.sh

Reply:
This would make the entrypoint to the docker image be bash. Do we want that ? I guess this image should has the entrypoint defined as celestia-appd. And maybe add a command `--help` after that for a default behaviour (even if it's not needed as by default, celestia-appd without arguments will return the help, it is more elegant, except that it adds an extra layer on top).
If I understand right, with this approach, we will be calling something like:
```ssh
$ docker run celestia-app celestia-appd <args>
```
which is redundant, opposed to the proposed approach:
```ssh
$ docker run celestia-app <args>
```
This is more like an executable container rather than a general purpose one that spawns a shell by default.
Example, geth are doing the same providing their binary as the entrypoint https://github.com/ethereum/go-ethereum/blob/master/Dockerfile.

Let me know what you think

Reply:
For the provided link https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint
As in the example, postgres uses an entrypoint script to do some initializations before  executing the command while allowing the users to run whatever they want. This comes from the fact that postgres has multiple binaries inside the image: `pg_virtualenv`, `pg_createcluster`, `pg_dropcluster` and many others.  Thus, allowing users to execute anything is a must.
In our case, it is only a single binary that we want to use in all cases...

Reply:
I sometimes want to `docker exec -it <container-name> /bin/bash` into the box and poke around. But I'm not strongly opposed to have a binary style docker image if it simplifies things for your use case.